sunpy/physics/differential_rotation.py

```diff
@@ -153,10 +150,15 @@ def rot_hpc(x, y, tstart, tend, frame_time='synodic', rot_type='howard', **kwarg
     by convert_hpc_hg.  This leads to very slightly different results from
     rot_hpc compared to rot_xy.
     """
+    x = coord.Tx
```

Comment:
these `Tx` `Ty` attributes are only valid if the `SkyCoord` frame is `Helioprojective.` If needs be you should transform to helioprojective. (I think this function converts *from* helioprojective to heliographic later, the transform should *only* be done if needed).